diff --git a/node_modules/expo-dev-launcher/ios/EXDevLauncherController.m b/node_modules/expo-dev-launcher/ios/EXDevLauncherController.m
index a31846c..1f2a0d5 100644
--- a/node_modules/expo-dev-launcher/ios/EXDevLauncherController.m
+++ b/node_modules/expo-dev-launcher/ios/EXDevLauncherController.m
@@ -699,7 +699,10 @@
 -(NSString *)getAppIcon
 {
   NSString *appIcon = @"";
-  NSString *appIconName = [[[[[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleIcons"] objectForKey:@"CFBundlePrimaryIcon"] objectForKey:@"CFBundleIconFiles"]  lastObject];
+  NSString *appIconName = nil;
+  @try {
+    appIconName = [[[[[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleIcons"] objectForKey:@"CFBundlePrimaryIcon"] objectForKey:@"CFBundleIconFiles"]  lastObject];
+  } @catch(NSException *_e) {}
 
   if (appIconName != nil) {
     NSString *resourcePath = [[NSBundle mainBundle] resourcePath];
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/DevLauncherViews.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/DevLauncherViews.swift
index 151e561..dd0b0cc 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/DevLauncherViews.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/DevLauncherViews.swift
@@ -111,9 +111,7 @@ struct RecentlyOpenedAppRow: View {
           .foregroundColor(.secondary)
       }
       .padding()
-#if !os(tvOS)
-      .background(Color(.secondarySystemBackground))
-#endif
+      .background(Color.expoSystemColors.secondarySystemBackground.color)
       .clipShape(RoundedRectangle(cornerRadius: 12))
     }
     .buttonStyle(PlainButtonStyle())
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/DevServerInfoModal.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/DevServerInfoModal.swift
index c30cc37..ca71a09 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/DevServerInfoModal.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/DevServerInfoModal.swift
@@ -28,9 +28,7 @@ struct DevServerInfoModal: View {
       info
     }
     .padding(20)
-    #if !os(tvOS)
-    .background(Color(.systemBackground))
-    #endif
+    .background(Color.expoSystemColors.systemBackground.color)
     .clipShape(RoundedRectangle(cornerRadius: 16))
     .shadow(color: .black.opacity(0.1), radius: 10, x: 0, y: 5)
     .padding(.horizontal, 40)
@@ -52,14 +50,6 @@ struct DevServerInfoModal: View {
     }
   }
 
-  #if os(tvOS)
-  let systemGray6 = Color(.systemGray)
-  let systemGray4 = Color(.systemGray)
-  #else
-  let systemGray6 = Color(.systemGray6)
-  let systemGray4 = Color(.systemGray4)
-  #endif
-
   private var info: some View {
     VStack(alignment: .leading, spacing: 12) {
       Text("Start a local development server with:")
@@ -70,11 +60,11 @@ struct DevServerInfoModal: View {
         .font(.system(.callout, design: .monospaced))
         .padding(.horizontal, 12)
         .padding(.vertical, 8)
-        .background(systemGray6)
+        .background(Color.expoSystemColors.systemGray6.color)
         .clipShape(RoundedRectangle(cornerRadius: 6))
         .overlay(
           RoundedRectangle(cornerRadius: 6)
-            .stroke(systemGray4, lineWidth: 1)
+            .stroke(Color.expoSystemColors.systemGray4.color, lineWidth: 1)
         )
 
       Text("Then, select the local server when it appears here.")
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/DevServersView.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/DevServersView.swift
index 42effcf..2472e38 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/DevServersView.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/DevServersView.swift
@@ -79,7 +79,7 @@ struct DevServersView: View {
         #if !os(tvOS)
           .overlay(
             RoundedRectangle(cornerRadius: 5)
-              .stroke(Color(.systemGray4), lineWidth: 1)
+              .stroke(Color.expoSystemColors.systemGray4.color, lineWidth: 1)
           )
         #endif
           .clipShape(RoundedRectangle(cornerRadius: 5))
@@ -89,9 +89,7 @@ struct DevServersView: View {
     }
     .animation(.easeInOut, value: showingURLInput)
     .padding()
-#if !os(tvOS)
-    .background(Color(showingURLInput ? .secondarySystemBackground : .systemBackground))
-#endif
+    .background(Color.expoSystemColors.secondarySystemBackground.color)
     .clipShape(RoundedRectangle(cornerRadius: 12))
   }
 
@@ -172,9 +170,7 @@ struct DevServerRow: View {
           .foregroundColor(.secondary)
       }
       .padding()
-#if !os(tvOS)
-      .background(Color(.secondarySystemBackground))
-#endif
+      .background(Color.expoSystemColors.secondarySystemBackground.color)
       .clipShape(RoundedRectangle(cornerRadius: 12))
     }
     .buttonStyle(PlainButtonStyle())
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/ErrorView.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/ErrorView.swift
index c8f4747..6f59224 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/ErrorView.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/ErrorView.swift
@@ -8,16 +8,6 @@ struct ErrorView: View {
   let onReload: () -> Void
   let onGoHome: () -> Void
 
-#if os(tvOS)
-  let systemGray6 = Color(.systemGray)
-  let systemGray4 = Color(.systemGray)
-  let systemBackground = Color(.white)
-#else
-  let systemGray6 = Color(.systemGray6)
-  let systemGray4 = Color(.systemGray4)
-  let systemBackground = Color(.systemBackground)
-#endif
-
   var body: some View {
     VStack(spacing: 0) {
       VStack(alignment: .leading, spacing: 12) {
@@ -38,7 +28,7 @@ struct ErrorView: View {
 
       actions
     }
-    .background(systemBackground)
+    .background(Color.expoSystemColors.systemBackground.color)
     .navigationBarHidden(true)
   }
 
@@ -59,7 +49,7 @@ struct ErrorView: View {
       .padding()
       .frame(maxWidth: .infinity, alignment: .leading)
     }
-    .background(systemGray6)
+    .background(Color.expoSystemColors.systemGray6.color)
     .cornerRadius(8)
   }
 
@@ -81,14 +71,12 @@ struct ErrorView: View {
           .foregroundColor(.black)
           .frame(maxWidth: .infinity)
           .padding()
-#if !os(tvOS)
-          .background(Color(.systemGray5))
-#endif
+          .background(Color.expoSystemColors.systemGray5.color)
           .cornerRadius(8)
       }
     }
     .padding(.horizontal, 20)
     .padding(.vertical, 20)
-    .background(systemBackground)
+    .background(Color.expoSystemColors.systemBackground.color)
   }
 }
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/HomeTabView.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/HomeTabView.swift
index fe20a86..d99c479 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/HomeTabView.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/HomeTabView.swift
@@ -65,9 +65,7 @@ struct HomeTabView: View {
         .padding()
     }
     .buttonStyle(PlainButtonStyle())
-    #if !os(tvOS)
-    .background(Color(.secondarySystemGroupedBackground))
-    #endif
+    .background(Color.expoSystemColors.secondarySystemGroupedBackground.color)
     .cornerRadius(18)
   }
 }
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/Navigation/Navigation.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/Navigation/Navigation.swift
index 1f339a5..dab885c 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/Navigation/Navigation.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/Navigation/Navigation.swift
@@ -58,9 +58,7 @@ struct DevLauncherNavigationHeader: View {
         } else {
           ZStack {
             Circle()
-            #if !os(tvOS)
-              .fill(Color(.systemGray6))
-            #endif
+              .fill(Color.expoSystemColors.systemGray6.color)
               .frame(width: 36, height: 36)
 
             Image("user-icon", bundle: getDevLauncherBundle())
@@ -72,9 +70,7 @@ struct DevLauncherNavigationHeader: View {
     }
     .padding(.horizontal)
     .padding(.vertical, 8)
-    #if !os(tvOS)
-    .background(Color(.systemBackground))
-    #endif
+    .background(Color.expoSystemColors.systemBackground.color)
   }
 
   @ViewBuilder
@@ -103,9 +99,7 @@ struct DevLauncherNavigationHeader: View {
           .aspectRatio(contentMode: .fill)
       } placeholder: {
         Circle()
-        #if !os(tvOS)
-          .fill(Color(.systemGray5))
-        #endif
+          .fill(Color.expoSystemColors.systemGray5.color)
           .overlay(
             Image(systemName: "person")
               .font(.system(size: 16))
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/SettingsTabView.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/SettingsTabView.swift
index f11995b..9dd343b 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/SettingsTabView.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/SettingsTabView.swift
@@ -68,9 +68,7 @@ struct SettingsTabView: View {
       Toggle("Show menu at launch", isOn: $viewModel.showOnLaunch)
     }
     .padding()
-#if !os(tvOS)
-    .background(Color(.secondarySystemBackground))
-#endif
+    .background(Color.expoSystemColors.secondarySystemBackground.color)
     .cornerRadius(12)
   }
 
@@ -118,9 +116,7 @@ struct SettingsTabView: View {
         }
         .padding()
       }
-#if !os(tvOS)
-      .background(Color(.secondarySystemBackground))
-#endif
+      .background(Color.expoSystemColors.secondarySystemBackground.color)
       .cornerRadius(12)
     }
   }
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/UpdatesTab/UpdatesListView.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/UpdatesTab/UpdatesListView.swift
index 9125c9f..701622a 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/UpdatesTab/UpdatesListView.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/UpdatesTab/UpdatesListView.swift
@@ -41,9 +41,7 @@ struct UpdatesListView: View {
         }
       }
     }
-    #if !os(tvOS)
-    .background(Color(.systemGroupedBackground))
-    #endif
+    .background(Color.expoSystemColors.systemGroupedBackground.color)
     .onChange(of: filterByCompatibility) { _ in
       applyFilters()
     }
diff --git a/node_modules/expo-dev-launcher/ios/SwiftUI/Utils/Utils.swift b/node_modules/expo-dev-launcher/ios/SwiftUI/Utils/Utils.swift
index b4a025c..2e11931 100644
--- a/node_modules/expo-dev-launcher/ios/SwiftUI/Utils/Utils.swift
+++ b/node_modules/expo-dev-launcher/ios/SwiftUI/Utils/Utils.swift
@@ -51,11 +51,7 @@ extension Text {
 
 extension View {
   func systemGroupedBackground() -> some View {
-    #if os(tvOS)
-    return self
-    #else
-    return self.background(Color(.systemGroupedBackground))
-    #endif
+    return self.background(Color.expoSystemColors.systemGroupedBackground.color)
   }
 }
 
