diff --git a/node_modules/react-native-bottom-tabs/ios/TabViewImpl.swift b/node_modules/react-native-bottom-tabs/ios/TabViewImpl.swift
index ad6de7d..2c9997f 100644
--- a/node_modules/react-native-bottom-tabs/ios/TabViewImpl.swift
+++ b/node_modules/react-native-bottom-tabs/ios/TabViewImpl.swift
@@ -26,7 +26,11 @@ struct TabViewImpl: View {
         renderTabItem(at: index)
       }
       .measureView(onLayout: { size in
+        #if os(tvOS)
+        onLayout(CGSize(width: 1920, height: 1080))
+        #else
         onLayout(size)
+        #endif
       })
     }
 #if !os(tvOS) && !os(macOS) && !os(visionOS)
@@ -294,6 +298,11 @@ extension View {
   func ignoresTopSafeArea(
     _ flag: Bool
   ) -> some View {
+    #if os(tvOS)
+    self
+     .ignoresSafeArea(.container, edges: .all)
+     .frame(idealWidth: 1920.0, idealHeight: 1080.0)
+    #else
     if flag {
       self
         .ignoresSafeArea(.container, edges: .vertical)
@@ -301,6 +310,7 @@ extension View {
       self
         .ignoresSafeArea(.container, edges: .bottom)
     }
+    #endif
   }
 
 #if !os(macOS)
