diff --git a/node_modules/react-native-bottom-tabs/ios/TabView/NewTabView.swift b/node_modules/react-native-bottom-tabs/ios/TabView/NewTabView.swift
index 229445f..e91daa1 100644
--- a/node_modules/react-native-bottom-tabs/ios/TabView/NewTabView.swift
+++ b/node_modules/react-native-bottom-tabs/ios/TabView/NewTabView.swift
@@ -39,7 +39,9 @@ struct NewTabView: AnyTabView {
                 labeled: props.labeled
               )
             }
+            #if !os(tvOS)
             .badge(tabData.badge.flatMap { !$0.isEmpty ? Text($0) : nil })
+            #endif
             .accessibilityIdentifier(tabData.testID ?? "")
           }
         }
